wrangler-storage/src/main/java/io/cdap/wrangler/dataset/schema/HealthcarePackageSchemaLoader.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.wrangler.dataset.schema;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import io.cdap.wrangler.dataset.schema.SchemaFile.MetaData;
+import io.cdap.wrangler.dataset.schema.SchemaFile.Standard;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Loads JSON schema from Google Cloud Healthcare API specific system artifact.
+ */
+public class HealthcarePackageSchemaLoader implements JSONSchemaLoader {
+
+  private static final String LOADER_TYPE = "Healthcare";
+  private static final String MANIFEST_FILE = "schemas/manifest.txt";
+
+  public HealthcarePackageSchemaLoader() {}
+
+  @Override
+  public String getLoaderType() {
+    return HealthcarePackageSchemaLoader.LOADER_TYPE;
+  }
+
+  @Override
+  public Collection<SchemaFileDescriptor> load() throws IOException {
+    List<SchemaFileDescriptor> descriptors = new ArrayList<>();
+    Gson gson = new Gson();
+    JsonParser jsonParser = new JsonParser();
+
+    InputStream manifest = HealthcarePackageSchemaLoader.class.getClassLoader()
+        .getResourceAsStream(HealthcarePackageSchemaLoader.MANIFEST_FILE);
+    BufferedReader br = new BufferedReader(new InputStreamReader(manifest));
+
+    while (br.ready()) {
+      String schemaFile = br.readLine();
+      InputStream schemaStream = HealthcarePackageSchemaLoader.class.getClassLoader()
+          .getResourceAsStream("schemas/" + schemaFile);
+
+      if (schemaStream != null) {
+        JsonElement root = jsonParser.parse(new BufferedReader(new InputStreamReader(schemaStream)));
+        SchemaFile definition = SchemaFile.builder()
+            .setJson(root.toString())
+            .build();
+        MetaData metadata = definition.getMetadata();
+        Standard standard = metadata.getStandard();
+        String id = String.format("%s_%s_%s", standard.getName(),
+            standard.getVersion(),
+            metadata.getVersion())
+            .toLowerCase()
+            .replace('.', '-')
+            .replaceAll("[^a-zA-Z0-9\\-_]", "");
```

Comment:
this can cause multiple schemas to end up with the same id. Is the manifest a standard format or is it something we decided on?

If we control the format, it should just be a json representation of a collection of SchemaDescriptors, with the SchemaStore just deserializing it directly with Gson instead of having custom logic.

Reply:
Updated so that the identifier in the manifest is the source of truth. The descriptor is now NamespacedId aware so I don't think deserializing directly with Gson is the best approach.